native/src/utils/renderHtml.ts

```diff
@@ -33,72 +54,191 @@ const renderJS = (cacheDictionary: Record<string, string>) => `
+      const container = getContainer(iframeContainer, 'iframe-info-text')
+      showMessage(text, container, iframeSource)
+      showSettingsButton(container)
+    }
+
+    function showOptIn(text, iframeContainer, source) {
+      function onClickHandler() {
+        window.ReactNativeWebView.postMessage(
+          JSON.stringify({ type: '${ALLOW_EXTERNAL_SOURCE_MESSAGE_TYPE}', source }),
+        )
+      }
+      
+      const container = getContainer(iframeContainer, 'iframe-info-text')
+      const checkbox = document.createElement('input')
+      checkbox.type = 'checkbox'
+      checkbox.name = 'opt-in-checkbox'
+      checkbox.id = checkbox.name
+      checkbox.onclick = onClickHandler
+      const label = document.createElement('label')
+      label.htmlFor = checkbox.name
+      showSource(label, source)
+      label.appendChild(document.createTextNode(text))
+      container.appendChild(label)
+      container.appendChild(checkbox)
+    }
+    
+    function handleSupportedIframeSources(iframe, iframeSource) {
+      const externalSourcePermissions = ${JSON.stringify(externalSourcePermissions)}
+      const iframeContainer = document.createElement('div')
+      iframeContainer.classList.add('iframe-container')
+      iframe.parentNode.appendChild(iframeContainer)
+      iframeContainer.appendChild(iframe)
+      // Scale the iframe height depending on device width and outside margin
+      const deviceWidth = '${deviceWidth}'
+      const pageContainerPadding = '${pageContainerPadding}'
+      const scaledHeight = (deviceWidth / Number(iframe.width)) * Number(iframe.height) - pageContainerPadding
+      iframe.setAttribute('height', scaledHeight)
+      if (externalSourcePermissions[iframeSource] === undefined) {
+        const message = '${t('remoteContent:knownResourceOptIn')}'
+        showOptIn(message, iframeContainer, iframeSource)
+        iframe.remove()
+      } else if (externalSourcePermissions[iframeSource]) {
+        // Add do not track parameter (only working for vimeo)
+        if (iframeSource === 'vimeo.com') {
+          const url = new URL(iframe.src)
+          url.searchParams.append('dnt', '1')
+          iframe.setAttribute('src', url.href)
+        }
+        const message = '${t('remoteContent:knownResourceContentMessage')}'
+        showMessageWithSettings(message, iframeContainer, iframeSource)
+      } else {
+        const message = '${t('remoteContent:knownResourceBlocked')}'
+        showMessageWithSettings(message, iframeContainer, iframeSource)
+        iframe.remove()
+      }
+    }
+
+    const iframes = document.querySelectorAll('iframe')
+    const supportedIframeSources = ${JSON.stringify(supportedIframeSources)}
+
+    iframes.forEach(iframe => {
+      const supportedIframeSource = supportedIframeSources.find(src => iframe.src.includes(src))
+      if (supportedIframeSource) {
+        handleSupportedIframeSources(iframe, supportedIframeSource)
+      } else {
+        iframe.remove()
```

Comment:
I think it would be better to show a short message, about whats going on. "A video was implemented here, but cannot be displayed due to content policies" or whatever.

Reply:
well i've implemented it like this before, but as discussed before with @steffenkleinle we decided not to make it more complicated than needed since the authors should be informed about which content can be embedded and which not. So this case shouldn't happen